wai-aria/role/invalid-roles.html

```diff
@@ -11,26 +11,109 @@
+<nav role=" " data-testname="nav with empty character as role" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="&#13" data-testname="nav with line break (&#13) character as role" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="&#9" data-testname="nav with tab (&#9) as role (should be treated as whitespace)" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="&#8203" data-testname="nav with zero-width space as role (should be treated as whitespace)" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="&#10240" data-testname="nav with braille space (10240) as role" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="&#x2800" data-testname="nav with braille space (x2800) as role" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="&nbsp" data-testname="nav with non-breaking space (nbsp) as role" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="&#20" data-testname="nav with standard space (nbsp) as role" data-expectedrole="navigation" class="ex">x</nav>
+
+<!-- Whitespace tests with <span> (including line breaks, tabs, zero-width space, braille space, non-breaking space, standard space) -->
+<span role=" " data-testname="span with empty character as role" class="ex-generic">x</span>
+<span role="&#13" data-testname="span with line break (&#13) character as role" class="ex-generic">x</span>
+<span role="&#9" data-testname="span with tab (&#9) as role (should be treated as whitespace)" class="ex-generic">x</span>
+<span role="&#8203" data-testname="span with zero-width space as role (should be treated as whitespace)" class="ex-generic">x</span>
+<span role="&#10240" data-testname="span with braille space (10240) as role" class="ex-generic">x</span>
+<span role="&#x2800" data-testname="span with braille space (x2800) as role" class="ex-generic">x</span>
+<span role="&nbsp" data-testname="span with non-breaking space (nbsp) as role" class="ex-generic">x</span>
+<span role="&#20" data-testname="span with standard space (nbsp) as role" class="ex-generic">x</span>
+
+<!-- Diacritics with <a> -->
+<a role="&#771" data-testname="link with role set to tilde diacritic" data-expectedrole="link" class="ex">x</a>
+<a role="foo&#771" data-testname="link with role foo that has tilde diacritic" data-expectedrole="link" class="ex">x</a>
+<a role="&#773" data-testname="link with role set to overline diacritic" data-expectedrole="link" class="ex">x</a>
+<a role="foo&#773" data-testname="link with role foo that has overline diacritic" data-expectedrole="link" class="ex">x</a>
+<a role="&#813" data-testname="link with role set to circumflex diacritic below" data-expectedrole="link" class="ex">x</a>
+<a role="foo&#813" data-testname="link with role foo that has circumflex diacritic below" data-expectedrole="link" class="ex">x</a>
+<a role="button&#771" data-testname="link with role set to button with tilde diacritic" data-expectedrole="link" class="ex">x</a>
```

Comment:
`data-expectedrole="link"` is wrong here b/c there is no `href="#"`... Either add it, or change these to `ex-generic`.

Reply:
PS. I didn't notice the error in review, but found it by checking the CI results.

Reply:
@cookiecrook Interesting, I'll add `href`s for these however, is the lack of href covered by another test? Just tested with Chrome and the a11y tree shows the explicit `role` for a `<a>` being honored even when the `href` is missing; macOS VoiceOver also announces the `role`. The lack of `href` definitely impacts keyboard operability but I was unaware it would negatively impact role calculation.

Reply:
You're right, these should map to `generic`! Here's the WAI-ARIA mapping for no href link however, it doesn't appear that WebKit/Blink currently conform: https://w3c.github.io/html-aam/#el-a-no-href.

Reply:
Already covered in https://github.com/web-platform-tests/wpt/blob/master/html-aam/roles-contextual.html#L20